### ISSUE TITLE
Implement FastText topics

### ISSUE THREAD

**Issue by lukewendling** [Sotera/watchman#171]
## Overview

Run a daily batch process that uses labeled tweets (def: a tweet with exactly 1 hashtag) to train a fasttext model that will classify all tweets during that day into 1 of X hashtags (topics). X == num. of most popular hashtags. Final output will group the tweets by QCR campaign.

## Process

1. Collect list of most popular hashtags for the period.
1. Train a fasttext model with all single hashtag tweets for the period that have a popular hashtag.
1. Classify all tweets for the period using the model, and include a confidence score.
1. Filter by results by confidence score to prune poor matches.
1. Group results by QCR campaign

**Comment by drJAGartner** [comment]
Yes, one note - the PM did more or less the same technique, but his nuance was to collect all tweets with individual tags and simply dedupe.  I think there's not a wrong way, but just note what you do for data cleansing so we can present it that way.

**Comment by lukewendling** [comment]
@drJAGartner 

## Process questions

1. Current jupyter nb excludes retweets in training. Should we do the same for quoted tweets?
1. Should we classify retweets, or ignore them in final output?

**Comment by lukewendling** [comment]
De-dupe: use 1 retweet as fill-in for original tweet.

**Comment by lukewendling** [comment]
fyi, de-duped retweets like this, mostly thanks to DataFrame#dropDuplicates():

```python
        df_retweets = df\
        .where('featurizer == "hashtag"')\
        .where('broadcast_post_id is not null')

        df_no_retweets = df\
        .where('featurizer == "hashtag"')\
        .where('broadcast_post_id == "null" or broadcast_post_id is null')

        df_retweets = df_retweets.dropDuplicates(['broadcast_post_id'])

        df_hash = df_retweets.union(df_no_retweets)
```
